src/core/api/thread_ftd_api.cpp

```diff
@@ -388,6 +388,13 @@ void otThreadSetThreadVersionCheckEnabled(otInstance *aInstance, bool aEnabled)
     AsCoreType(aInstance).Get<Mle::MleRouter>().SetThreadVersionCheckEnabled(aEnabled);
 }
 
+#if OPENTHREAD_CONFIG_PLATFORM_UDP_ENABLE
+void otThreadSetTmfHostCheckEnabled(otInstance *aInstance, bool aEnabled)
+{
+    AsCoreType(aInstance).Get<Ip6::Ip6>().SetTmfHostCheckEnabled(aEnabled);
```

Comment:
the function called `SetTmfHostCheckEnabled` (presume the one in `ip6.cpp` / `ip6.hpp`) is only available if `OPENTHREAD_CONFIG_REFERENCE_DEVICE_ENABLE` is set. What if this build flag is cleared - wouldn't there be a risk of build error because this function is not found?